stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/state/internal/ChatClientStateCalls.kt

```diff
@@ -27,18 +27,29 @@ import io.getstream.chat.android.offline.plugin.state.channel.ChannelState
 import io.getstream.chat.android.offline.plugin.state.channel.thread.ThreadState
 import io.getstream.chat.android.offline.plugin.state.querychannels.QueryChannelsState
 import kotlinx.coroutines.CoroutineScope
+import java.util.Date
+
+private const val MAX_TIME_OF_STATE_MILI = 300
 
 /**
  * Adapter for [ChatClient] that wraps some of it's request.
  */
-internal class ChatClientStateCalls(
+internal class ChatClientStateCalls internal constructor(
     private val chatClient: ChatClient,
     private val state: StateRegistry,
-    private val scope: CoroutineScope
+    private val scope: CoroutineScope,
 ) {
+
+    private val requestTimeMap: MutableMap<Int, Date> = mutableMapOf()
     /** Reference request of the channels query. */
-    internal fun queryChannels(request: QueryChannelsRequest): QueryChannelsState {
-        chatClient.queryChannels(request).launch(scope)
+    internal fun queryChannels(request: QueryChannelsRequest, forceRefresh: Boolean): QueryChannelsState {
+        val queryHashCode = request.hashCode()
+
+        if (isStateOld(queryHashCode) || forceRefresh) {
+            chatClient.queryChannels(request).launch(scope)
+            requestTimeMap[queryHashCode] = Date()
+        }
```

Comment:
I'm curious if we should protect this block with `mutex` since we use a regular map.
Not sure if `queryChannels` is gonna be invoked from different threads.

Reply:
What about using a `ConcurrentHashMap`?